src/cam_qhy.cpp

```diff
@@ -572,14 +935,29 @@ bool Camera_QHY::Capture(int duration, usImage& img, int options, const wxRect&
         return true;
     }
 # endif
-    if (ret == QHYCCD_SUCCESS)
+
+    /*
+     * Guidance regarding exposure handling. From QHY, October 2024
+     *
+     * The return code of ExpQHYCCDSingleFrame() must be considered with the exposure length
+     * to determine when to call GetQHYCCDSingleFrame().
+     *
+     * If return code == QHYCCD_READ_DIRECTLY && exposure duration < 3s, call GetQHYCCDSingleFrame() immediately.
+     * If return code == QHYCCD_READ_DIRECTLY && exposure duration >= 3s, call GetQHYCCDSingleFrame() after 2s.
+     * If return code != QHYCCD_READ_DIRECTLY, call GetQHYCCDSingleFrame() at the end of the exposure duration.
+     */
+    if (ret == QHYCCD_READ_DIRECTLY)
     {
-        Debug.Write(wxString::Format("QHY: 200ms delay needed\n"));
-        WorkerThread::MilliSleep(200);
+        if (duration > 3000)
+        {
+            Debug.Write(wxString::Format("QHY: 2000ms delay needed\n"));
+            WorkerThread::MilliSleep(2000);
+        }
     }
-    if (ret == QHYCCD_READ_DIRECTLY)
+    else
     {
-        // Debug.Write("QHYCCD_READ_DIRECTLY\n");
+        Debug.Write(wxString::Format("QHY: %dms delay needed\n", duration));
```

Comment:
optional: maybe remove this? may be a little excessive logging this especially for small durations where the logging it self could even take longer than the requested delay

Reply:
Definitely. I had that there to follow what was there previously as well as to observe how different camera models did their return codes for the exposure. While neat, it's certainly not needed.